ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -96,9 +105,31 @@ constructor(
       }
       .stateIn(viewModelScope, SharingStarted.Lazily, null)
 
+  val tileOverlays: LiveData<List<TileSource>>
+  val mbtilesFilePaths: LiveData<Set<String>>
+  val offlineImageryEnabled: Flow<Boolean> = mapStateRepository.offlineImageryFlow
+
   init {
     cameraUpdateRequests = mapController.getCameraUpdates().map { Event.create(it) }.toLiveData()
     mapType = mapStateRepository.mapTypeFlowable.toLiveData()
+    tileOverlays =
+      surveyRepository.activeSurveyFlow
+        .mapNotNull { it?.tileSources?.mapNotNull(this::toLocalTileSource) ?: listOf() }
+        .asLiveData()
+    mbtilesFilePaths =
+      offlineAreaRepository
+        .downloadedTileSetsOnceAndStream()
+        .map { set: Set<MbtilesFile> -> set.map(MbtilesFile::path).toSet() }
+        .toLiveData()
+  }
+
+  // TODO: Maybe create a new data class object which is not of type TileSource.
```

Comment:
Please create and add bug number to track.

Reply:
Done